src/dotnet/commands/dotnet-tool/uninstall/ToolUninstallLocalCommand.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ToolManifest;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Tools.Tool.Uninstall
+{
+    internal class ToolUninstallLocalCommand : CommandBase
+    {
+        private readonly IToolManifestFinder _toolManifestFinder;
+        private readonly IToolManifestEditor _toolManifestEditor;
+        private readonly IReporter _reporter;
+
+        private readonly PackageId _packageId;
+        private readonly string _explicitManifestFile;
+
+        public ToolUninstallLocalCommand(
+            AppliedOption appliedCommand,
+            ParseResult parseResult,
+            IToolManifestFinder toolManifestFinder = null,
+            IToolManifestEditor toolManifestEditor = null,
+            IReporter reporter = null)
+            : base(parseResult)
+        {
+            if (appliedCommand == null)
+            {
+                throw new ArgumentNullException(nameof(appliedCommand));
+            }
+
+            _packageId = new PackageId(appliedCommand.Arguments.Single());
+            _explicitManifestFile = appliedCommand.SingleArgumentOrDefault("--tool-manifest");
+
+            _reporter = (reporter ?? Reporter.Output);
+
+            _toolManifestFinder = toolManifestFinder ??
+                                  new ToolManifestFinder(new DirectoryPath(Directory.GetCurrentDirectory()));
+            _toolManifestEditor = toolManifestEditor ?? new ToolManifestEditor();
+        }
+
+        public override int Execute()
+        {
+            var manifestFile = string.IsNullOrWhiteSpace(_explicitManifestFile)
+                ? _toolManifestFinder.FindFirst()
+                : new FilePath(_explicitManifestFile);
+
+            _toolManifestEditor.Remove(manifestFile, _packageId);
```

Comment:
If there anything we can do to remove the tool package from the cache?  As a user, I'd expect that `dotnet tool uninstall --local` to delete the package on disk (similar to `dotnet tool uninstall --global`) and also remove it from the manifest.

Reply:
in local tools we use nuget cache instead of directly manage files, so it should be delete by nuget (use dotnet nuget command to clean it). It will also be problematic when there is another manifest has the same tool, we would make it not available since local tools cache are shared.

Reply:
That is unfortunate.  Personally I don't like using global locations (without refcounts) for repo-level stuff because it's not easy to "blast" away.